exp/controllers/awsmachinepool_controller.go

```diff
@@ -298,6 +298,21 @@ func (r *AWSMachinePoolReconciler) reconcileNormal(ctx context.Context, machineP
 		return nil
 	}
 
+	lifecycleHookScope, err := scope.NewLifecycleHookScope(scope.LifecycleHookScopeParams{
+		Client:         r.Client,
+		Logger:         &machinePoolScope.Logger,
+		MachinePool:    machinePoolScope.MachinePool,
+		AWSMachinePool: machinePoolScope.AWSMachinePool,
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to create lifecycle hook scope")
+	}
+
+	if err := reconSvc.ReconcileLifecycleHooks(*lifecycleHookScope, asgsvc); err != nil {
+		r.Recorder.Eventf(machinePoolScope.AWSMachinePool, corev1.EventTypeWarning, "FaileLifecycleHooksReconcile", "Failed to reconcile lifecycle hooks: %v", err)
```

Comment:
That's an error, not a warning

Reply:
There's no "EventTypeWarning", this is how the other reconciles push events to the MachinePool (see line 284 for the ReconcileLaunchTemplate).
We push an event to the MachinePool so that the users can see the events by looking at the object, and exit the reconciliation loop by throwing an error